src/components/v2/MintRepayVai/Mint/index.tsx

```diff
@@ -0,0 +1,127 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import BigNumber from 'bignumber.js';
+import Typography from '@mui/material/Typography';
+
+import { convertWeiToCoins } from 'utilities/common';
+import { Icon } from '../../Icon';
+import { TokenTextField } from '../../TokenTextField';
+import { SecondaryButton } from '../../Button';
+import { useStyles } from './styles';
+
+const VAI_SYMBOL = 'vai';
+
+export interface IMintUiProps {
+  disabled: boolean;
+  isSubmitting: boolean;
+  onSubmit: (value: BigNumber) => Promise<void>;
+  limitWei: BigNumber;
+  mintFeePercentage: number;
+}
+
+// TODO: Move to dashboard component/container once created
+export const MintUi: React.FC<IMintUiProps> = ({
+  disabled,
+  limitWei,
+  mintFeePercentage,
+  isSubmitting,
+  onSubmit,
+}) => {
+  const styles = useStyles();
+  const [value, setValue] = React.useState<BigNumber | ''>('');
+
+  const isValueValid = value && value.isGreaterThan(0);
+
+  // Convert limit into VAI
+  const readableLimitVai = React.useMemo(
+    () => convertWeiToCoins({ value: limitWei, tokenSymbol: VAI_SYMBOL }).toString(),
+    [limitWei],
+  );
+
+  // Calculate fee
+  const feeWei = new BigNumber(value || 0).multipliedBy(mintFeePercentage).dividedBy(100);
+  const readableFeeVai = convertWeiToCoins({ value: feeWei, tokenSymbol: VAI_SYMBOL }).toString();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    // Prevent native submission behavior
+    e.preventDefault();
+
+    if (value) {
+      await onSubmit(value);
+
+      // Reset value
+      setValue('');
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
```

Comment:
We should create a form component that standardizes our form behaviors and layout such validation and form level errors

Reply:
I liked using Formik, but am happy with whatever works for us.

Reply:
I think ti will be the best option

Reply:
Done